src/transformers/integrations/integration_utils.py

```diff
@@ -1263,14 +1277,15 @@ def _ensure_at_least_run_without_monitoring(self):
             self._use_initial_run()
         else:
             if not self._run:
-                self._initialize_run(
-                    with_id=self._run_id,
-                    capture_stdout=False,
-                    capture_stderr=False,
-                    capture_hardware_metrics=False,
-                    capture_traceback=False,
-                )
-                self._is_monitoring_run = False
+                with preserve_neptune_kwargs(callback=self):
```

Comment:
I would recommend to override the params as a part of `def _initialize_run`:
```python
    def _initialize_run(self, **additional_neptune_kwargs):
        ...

        self._stop_run_if_exists()

        try:
            run_params = additional_neptune_kwargs.copy()
            run_params.update(self._init_run_kwargs)
            self._run = init_run(**run_params)
            self._run_id = self._run["sys/id"].fetch()
        except (NeptuneMissingProjectNameException, NeptuneMissingApiTokenException) as e:
            raise NeptuneMissingConfiguration() from e
```

I'm not sure if I haven't done any typo etc. but I don't see a need to "preserve" these params.

Reply:
Actually by use case mentioned here: https://github.com/neptune-ai/neptune-client/issues/1663 let's override with the user-provided parameters.

Reply:
But setting those `capture_*` to `False` is only required in one invocation of `_initialize_run`. I'm not sure I understand